src/DistributedArrays.jl

```diff
@@ -487,4 +487,21 @@ Base.scale!(A::DArray, x::Number) = begin
     return A
 end
 
+Base.sum(d::DArray) = reduce(Base.AddFun(), d)
+Base.maximum(d::DArray) = reduce(Base.MaxFun(), d)
+Base.minimum(d::DArray) = reduce(Base.MinFun(), d)
+Base.maxabs(d::DArray) = maximum([fetch(r) for r in [@spawnat p maxabs(localpart(d)) for p in procs(d)]])
```

Comment:
Changing the `mapreduce` signature causes an ambiguity warning:

```
Warning: New definition 
    mapreduce(Any,Any,DistributedArrays.DArray{T,N,A}) at /home/amitm/.julia/v0.4/DistributedArrays/src/DistributedArrays.jl:415
is ambiguous with: 
    mapreduce(Any,Function,AbstractArray{T,N}) at reduce.jl:156.
```

I suggest leaving it as is.